src/header/types/accept.rs

```diff
@@ -0,0 +1,325 @@
+    /// assert_eq!(
+    ///     Accept::decode(&mut raw_header.iter()).unwrap(),
+    ///     Some(typed_header)
+    /// ); 
+    /// ```
+    fn decode<'header, Iter>(values: &mut Iter) -> Result<Option<Self>, Self::DecodeError>
+    where
+        Iter: Iterator<Item = &'header HeaderValue>,
+    {
+        let mut accept = LinkedHashSet::new();
+        let mut present = false;
+        for value in values {
+            let media = value.as_str().split(',');
+            for mtype in media {
+                let media_type = MediaType::try_from(syntax::trim_whitespace(mtype))?;
+                accept.insert(media_type);
+            }
+            present = true;
+        }
+        if present {
+            Ok(Some(Accept(accept)))
+        } else {
+            Ok(None)
+        }
+    }
+    
+    /// Converts the [`Accept`] type to raw header values.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::convert::TryFrom;
+    ///
+    /// use rtsp::header::map::TypedHeader;
+    /// use rtsp::header::types::accept::MediaType;
+    /// use rtsp::header::types::Accept;
+    /// use rtsp::header::value::HeaderValue;
+    /// use rtsp::header::types::accept::QualityParam;
+    /// use mime::*;
+    /// use std::str::FromStr;
+    /// 
+    /// let typed_header = vec![MediaType::new(mime::STAR_STAR, None), MediaType::new(Mime::from_str("video/*").unwrap(), Some(QualityParam::new(0.5)))]
+    ///     .into_iter()
+    ///     .collect::<Accept>();
+    /// let expected_raw_headers = vec![
+    ///     vec![HeaderValue::try_from("*/*, video/* ;q=0.5").unwrap()],
+    ///     vec![HeaderValue::try_from("video/* ;q=0.5, */*").unwrap()],
+    /// ];
+    /// let mut raw_header = vec![];
+    /// typed_header.encode(&mut raw_header);
+    /// assert!(raw_header == expected_raw_headers[0] ||
+    ///         raw_header == expected_raw_headers[1]);
+    /// ```
+    fn encode<Target>(&self, values: &mut Target)
+    where
+        Target: Extend<HeaderValue>
+    {
+        // Unsafe Justification
+        //
+        // Header values must be valid UTF-8, and since we know that the [`RangeFormat`] type
+        // guarantees valid ASCII-US (with no newlines), it satisfies the constraints.
+        let value = self.iter().map(MediaType::as_str).join(", ");
+        values.extend(once(unsafe { HeaderValue::from_string_unchecked(value) }));    }
+
+    fn header_name() -> &'static HeaderName
+    where   
+        Self: Sized
+    {
+        &HeaderName::Accept
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct MediaType{
+    m_type: Mime,
+    quality: Option<QualityParam>
+
+}
+
+impl Default for MediaType {
+
+    fn default() -> Self {
+        MediaType{
+            m_type: mime::TEXT_PLAIN,
+            quality: None
+        }
+    }
+}
+
+impl Eq for MediaType {}
+
+impl Hash for MediaType {
+
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.m_type.hash(state);
```

Comment:
We should hash the quality too, so I think this should be derived. It also doesn't worry correctly at present as the equality behavior is different from the hashing behavior (you check quality in equality, but you don't hash it).